src/commands/convert.ts

```diff
@@ -0,0 +1,69 @@
+import { promises as fPromises } from 'fs';
+import { flags } from '@oclif/command';
+import * as parser from '@asyncapi/parser';
+import Command from '../base';
+import { ValidationError } from '../errors/validation-error';
+import { load } from '../models/SpecificationFile';
+import { SpecificationFileNotFound } from '../errors/specification-file';
+import { convert } from '@asyncapi/converter';
+
+export default class Convert extends Command {
+  static description = 'convert asyncapi documents older to newer versions';
+
+  static flags = {
+    help: flags.help({ char: 'h' }),
+    file: flags.string({ char: 'f', description: 'path to the file to convert' }),
+    output: flags.string({ char: 'o', description: 'path to the file where the result is saved' }),
+    'target-version': flags.string({ char: 't', description: 'asyncapi version to convert to', default: '2.3.0' }),
+    id: flags.string({ char: 'i', description: 'application id, if needed' })
+  }
+
+  static args = [
+    { name: 'spec-file', description: 'spec path, url, or context-name', required: false },
+  ]
+
+  async run() {
+    const { args, flags } = this.parse(Convert);
+    const filePath = args['spec-file'];
+    let specFile;
+    let convertedFile;
+
+    try {
+      // LOAD FILE
+      specFile = await load(filePath);
+
+      // VALIDATION
+      if (specFile.getFilePath()) {
+        await parser.parse(specFile.text());
+        this.log(`File ${specFile.getFilePath()} successfully validated!`);
+      } else if (specFile.getFileURL()) {
+        await parser.parse(specFile.text());
+        this.log(`URL ${specFile.getFileURL()} successfully validated!`);
+      }
```

Comment:
Here you have problem with that `Error: Version 1.2.0 is not supported` error :) ParserJS doesn't support `1.X.X` versions (including `2.0.0-rcX`), so you should first check if given version is supported and then make the validation process, if not you should only convert file :)

> Does it make sense to keep using this flag? Or should we consider to remove it from convert command.

I think that we should rid of that flag. Using this flag in the converter itself is quite "problematic", because you can always add the `id` field to the spec manually after the conversion, so I would get rid of it. cc @fmvilas

> For 2.x versions: The conversion is successful but the --id flag applied doesn't have any effect on the converted file.

Because that flag adds the `id` field to the converted file (from `1.X.X`), we don't change `id` between `2.X.X` -> `2.X.X` conversion.

Reply:
@magicmatatjahu @fmvilas 
Should we skip then the validation process for every version to avoid complexity and just convert the file? Do you agree?

Reply:
@peter-rr Atm converting `2.X.X` -> `2.X.X` only bumps the version of AsyncAPI, so we can say that even invalid spec, we will perform converting without problems. Of course we cannot validate the `1.X.X` versions. So... please don't validate any spec, only convert, hah 😄

Reply:
Validation process removed  as well as `--id` flag. Everything seems to be working properly 👍